pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,438 @@
+type component struct {
+	config *Config
+}
+
+// Config contains all the config information EgressGateway needs to render component.
+type Config struct {
+	PullSecrets  []*corev1.Secret
+	Installation *operatorv1.InstallationSpec
+	OSType       rmeta.OSType
+	EgressGW     *operatorv1.EgressGateway
+
+	egwImage          string
+	EgressGWVxlanVNI  int
+	EgressGWVxlanPort int
+
+	Openshift bool
+	// Whether or not the cluster supports pod security policies.
+	UsePSP bool
+}
+
+func (c *component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.config.Installation.Registry
+	path := c.config.Installation.ImagePath
+	prefix := c.config.Installation.ImagePrefix
+
+	if c.config.OSType != c.SupportedOSType() {
+		return fmt.Errorf("Egress Gateway is supported only on %s", c.SupportedOSType())
+	}
+
+	var err error
+	c.config.egwImage, err = components.GetReference(components.ComponentEgressGateway, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *component) Objects() ([]client.Object, []client.Object) {
+	objects := []client.Object{}
+	objects = append(objects, c.egwServiceAccount())
+	objects = append(objects, c.egwDeployment())
+	if c.config.UsePSP {
+		objects = append(objects, c.egwPodSecurityPolicy())
+		objects = append(objects, c.egwClusterRole())
+		objects = append(objects, c.egwClusterRoleBinding())
+	} else if c.config.Openshift {
+		objects = append(objects, c.egwSecurityContextConstraints())
+	}
+	return objects, nil
+}
+
+func (c *component) Ready() bool {
+	return true
+}
+
+func (c *component) egwDeployment() *appsv1.Deployment {
+	d := appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.config.EgressGW.Name,
+			Namespace: c.config.EgressGW.Namespace,
+			Labels:    c.config.EgressGW.Spec.Template.Metadata.Labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: c.config.EgressGW.Spec.Replicas,
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": c.config.EgressGW.Name}},
+			Template: *c.deploymentPodTemplate(),
+		},
+	}
+	return &d
+}
+
+func (c *component) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range c.config.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: c.egwBuildAnnotations(),
+			Labels:      c.config.EgressGW.Spec.Template.Metadata.Labels,
+		},
+		Spec: corev1.PodSpec{
+			ImagePullSecrets:              ps,
+			Affinity:                      c.config.EgressGW.GetAffinity(),
+			TopologySpreadConstraints:     c.config.EgressGW.GetTopoConstraints(),
+			NodeSelector:                  c.config.EgressGW.GetNodeSelector(),
+			TerminationGracePeriodSeconds: c.config.EgressGW.GetTerminationGracePeriod(),
```

Comment:
You don't need to put these here.

Instead, you should call this function so that the overrides machinery handles this for you: https://github.com/tigera/operator/blob/74f866faa72aa7be30583b207c7a05fc26540809/pkg/render/typha.go#L429-L431

This will make EGW automatically pick up any new fields that get added in the future. 

Please take a look at how the Typha render logic does this today and model EGW after it.